### DISCUSSION TITLE
Deploy---> alex---> dev---> DataPipeline---> DruidIngestion

### DISCUSSION THREAD

**Discussion by srikanth-6520** [Sunbird-Ed/Community#528] [Bugs] [answered]
Hi Team,
we are getting an error, while deploying the jenkins job , below we have provided jenkins console.

TASK [druid-ingestion : Override the schema spec with context and  Copy schema file] ***
task path: /var/lib/jenkins/workspace/Deploy/alex/dev/DataPipeline/DruidIngestion/ansible/roles/druid-ingestion/tasks/main.yml:12
ok: [172.16.1.193] => (item=raw_offline_desktop) => {
    "ansible_loop_var": "item",
    "changed": false,
    "checksum": "3b5746f591cd2af70b2e078152820346d907885b",
    "dest": "/home/druid/raw_offline_desktop.json",
    "gid": 1002,
    "group": "druid",
    "item": "raw_offline_desktop",
    "mode": "0644",
    "owner": "druid",
    "path": "/home/druid/raw_offline_desktop.json",
    "size": 54330,
    "state": "file",
    "uid": 1002
}
ok: [172.16.1.193] => (item=raw_summary_events) => {
    "ansible_loop_var": "item",
    "changed": false,
    "checksum": "8301c34200e0716c002408502f004a6045183b27",
    "dest": "/home/druid/raw_summary_events.json",
    "gid": 1002,
    "group": "druid",
    "item": "raw_summary_events",
    "mode": "0644",
    "owner": "druid",
    "path": "/home/druid/raw_summary_events.json",
    "size": 49213,
    "state": "file",
    "uid": 1002
}
ok: [172.16.1.193] => (item=raw_telemetry_events) => {
    "ansible_loop_var": "item",
    "changed": false,
    "checksum": "809258a4f83250e1eb22cfcc55452eee38b8c61d",
    "dest": "/home/druid/raw_telemetry_events.json",
    "gid": 1002,
    "group": "druid",
    "item": "raw_telemetry_events",
    "mode": "0644",
    "owner": "druid",
    "path": "/home/druid/raw_telemetry_events.json",
    "size": 63243,
    "state": "file",
    "uid": 1002
}
ok: [172.16.1.193] => (item=raw_telemetry_feedback_events) => {
    "ansible_loop_var": "item",
    "changed": false,
    "checksum": "859a0efe6a50388cfd8302051d08d99a029b6ecb",
    "dest": "/home/druid/raw_telemetry_feedback_events.json",
    "gid": 1002,
    "group": "druid",
    "item": "raw_telemetry_feedback_events",
    "mode": "0644",
    "owner": "druid",
    "path": "/home/druid/raw_telemetry_feedback_events.json",
    "size": 34016,
    "state": "file",
    "uid": 1002
}
ok: [172.16.1.193] => (item=rollup_error_hourly_syncts) => {
    "ansible_loop_var": "item",
    "changed": false,
    "checksum": "b3571426fc0c9cb2944e72a8ac087175439e23ae",
    "dest": "/home/druid/rollup_error_hourly_syncts.json",
    "gid": 1002,
    "group": "druid",
    "item": "rollup_error_hourly_syncts",
    "mode": "0644",
    "owner": "druid",
    "path": "/home/druid/rollup_error_hourly_syncts.json",
    "size": 11996,
    "state": "file",
    "uid": 1002
}
ok: [172.16.1.193] => (item=rollup_summary_distinct_counts) => {
    "ansible_loop_var": "item",
    "changed": false,
    "checksum": "5d1d7933ad3063977c1fabd16f9c5d7bbe6c7fbb",
    "dest": "/home/druid/rollup_summary_distinct_counts.json",
    "gid": 1002,
    "group": "druid",
    "item": "rollup_summary_distinct_counts",
    "mode": "0644",
    "owner": "druid",
    "path": "/home/druid/rollup_summary_distinct_counts.json",
    "size": 14437,
    "state": "file",
    "uid": 1002
}
ok: [172.16.1.193] => (item=rollup_summary_syncts) => {
    "ansible_loop_var": "item",
    "changed": false,
    "checksum": "9be0b06466f35838a7ed8c8c62b4a76b855289c6",
    "dest": "/home/druid/rollup_summary_syncts.json",
    "gid": 1002,
    "group": "druid",
    "item": "rollup_summary_syncts",
    "mode": "0644",
    "owner": "druid",
    "path": "/home/druid/rollup_summary_syncts.json",
    "size": 14812,
    "state": "file",
    "uid": 1002
}
ok: [172.16.1.193] => (item=rollup_telemetry_audit_syncts) => {
    "ansible_loop_var": "item",
    "changed": false,
    "checksum": "a619122db56bfa6bb7526aca5bca3402f5e4cc8d",
    "dest": "/home/druid/rollup_telemetry_audit_syncts.json",
    "gid": 1002,
    "group": "druid",
    "item": "rollup_telemetry_audit_syncts",
    "mode": "0644",
    "owner": "druid",
    "path": "/home/druid/rollup_telemetry_audit_syncts.json",
    "size": 26659,
    "state": "file",
    "uid": 1002
}
ok: [172.16.1.193] => (item=rollup_telemetry_hourly_syncts) => {
    "ansible_loop_var": "item",
    "changed": false,
    "checksum": "317b5f79fa7d0519ddf6332e715c136218955d8e",
    "dest": "/home/druid/rollup_telemetry_hourly_syncts.json",
    "gid": 1002,
    "group": "druid",
    "item": "rollup_telemetry_hourly_syncts",
    "mode": "0644",
    "owner": "druid",
    "path": "/home/druid/rollup_telemetry_hourly_syncts.json",
    "size": 15821,
    "state": "file",
    "uid": 1002
}
ok: [172.16.1.193] => (item=rollup_telemetry_syncts) => {
    "ansible_loop_var": "item",
    "changed": false,
    "checksum": "16b869869300e1bf6c1b1d19a60c742bb8d7837b",
    "dest": "/home/druid/rollup_telemetry_syncts.json",
    "gid": 1002,
    "group": "druid",
    "item": "rollup_telemetry_syncts",
    "mode": "0644",
    "owner": "druid",
    "path": "/home/druid/rollup_telemetry_syncts.json",
    "size": 15573,
    "state": "file",
    "uid": 1002
}
ok: [172.16.1.193] => (item=rollup_tpd_hourly_syncts) => {
    "ansible_loop_var": "item",
    "changed": false,
    "checksum": "21e06e29e362de5cd44d18d087f996ba4eed73dd",
    "dest": "/home/druid/rollup_tpd_hourly_syncts.json",
    "gid": 1002,
    "group": "druid",
    "item": "rollup_tpd_hourly_syncts",
    "mode": "0644",
    "owner": "druid",
    "path": "/home/druid/rollup_tpd_hourly_syncts.json",
    "size": 5713,
    "state": "file",
    "uid": 1002
}

TASK [druid-ingestion : Start kafka supervisor tasks for defined datasources] ***
task path: /var/lib/jenkins/workspace/Deploy/alex/dev/DataPipeline/DruidIngestion/ansible/roles/druid-ingestion/tasks/main.yml:21
[WARNING]: Consider using the get_url or uri module rather than running 'curl'.
If you need to use command because get_url or uri is insufficient you can add
'warn: false' to this command task or set 'command_warnings=False' in
ansible.cfg to get rid of this message.

failed: [172.16.1.193] (item=raw_offline_desktop) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": "curl -X POST -H \"Content-Type:application/json\" -d @/home/druid/raw_offline_desktop.json [http://172.16.1.193:8090/druid/indexer/v1/supervisor",](http://172.16.1.193:8090/druid/indexer/v1/supervisor%22,%1B[0m)
    "delta": "0:00:00.015634",
    "end": "2023-06-02 07:18:32.880389",
    "item": "raw_offline_desktop",
    "rc": 7,
    "start": "2023-06-02 07:18:32.864755"
}

STDERR:

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 172.16.1.193 port 8090: Connection refused

MSG:

non-zero return code

failed: [172.16.1.193] (item=raw_summary_events) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": "curl -X POST -H \"Content-Type:application/json\" -d @/home/druid/raw_summary_events.json [http://172.16.1.193:8090/druid/indexer/v1/supervisor",](http://172.16.1.193:8090/druid/indexer/v1/supervisor%22,%1B[0m)
    "delta": "0:00:00.020362",
    "end": "2023-06-02 07:18:33.158742",
    "item": "raw_summary_events",
    "rc": 7,
    "start": "2023-06-02 07:18:33.138380"
}

STDERR:

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 172.16.1.193 port 8090: Connection refused

MSG:

non-zero return code

failed: [172.16.1.193] (item=raw_telemetry_events) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": "curl -X POST -H \"Content-Type:application/json\" -d @/home/druid/raw_telemetry_events.json [http://172.16.1.193:8090/druid/indexer/v1/supervisor",](http://172.16.1.193:8090/druid/indexer/v1/supervisor%22,%1B[0m)
    "delta": "0:00:00.012740",
    "end": "2023-06-02 07:18:33.396945",
    "item": "raw_telemetry_events",
    "rc": 7,
    "start": "2023-06-02 07:18:33.384205"
}

STDERR:

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 172.16.1.193 port 8090: Connection refused

MSG:

non-zero return code

failed: [172.16.1.193] (item=raw_telemetry_feedback_events) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": "curl -X POST -H \"Content-Type:application/json\" -d @/home/druid/raw_telemetry_feedback_events.json [http://172.16.1.193:8090/druid/indexer/v1/supervisor",](http://172.16.1.193:8090/druid/indexer/v1/supervisor%22,%1B[0m)
    "delta": "0:00:00.015061",
    "end": "2023-06-02 07:18:33.640869",
    "item": "raw_telemetry_feedback_events",
    "rc": 7,
    "start": "2023-06-02 07:18:33.625808"
}

STDERR:

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 172.16.1.193 port 8090: Connection refused

MSG:

non-zero return code

failed: [172.16.1.193] (item=rollup_error_hourly_syncts) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": "curl -X POST -H \"Content-Type:application/json\" -d @/home/druid/rollup_error_hourly_syncts.json [http://172.16.1.193:8090/druid/indexer/v1/supervisor",](http://172.16.1.193:8090/druid/indexer/v1/supervisor%22,%1B[0m)
    "delta": "0:00:00.018005",
    "end": "2023-06-02 07:18:33.921969",
    "item": "rollup_error_hourly_syncts",
    "rc": 7,
    "start": "2023-06-02 07:18:33.903964"
}

STDERR:

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 172.16.1.193 port 8090: Connection refused

MSG:

non-zero return code

failed: [172.16.1.193] (item=rollup_summary_distinct_counts) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": "curl -X POST -H \"Content-Type:application/json\" -d @/home/druid/rollup_summary_distinct_counts.json [http://172.16.1.193:8090/druid/indexer/v1/supervisor",](http://172.16.1.193:8090/druid/indexer/v1/supervisor%22,%1B[0m)
    "delta": "0:00:00.015254",
    "end": "2023-06-02 07:18:34.156187",
    "item": "rollup_summary_distinct_counts",
    "rc": 7,
    "start": "2023-06-02 07:18:34.140933"
}

STDERR:

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 172.16.1.193 port 8090: Connection refused

MSG:

non-zero return code

failed: [172.16.1.193] (item=rollup_summary_syncts) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": "curl -X POST -H \"Content-Type:application/json\" -d @/home/druid/rollup_summary_syncts.json [http://172.16.1.193:8090/druid/indexer/v1/supervisor",](http://172.16.1.193:8090/druid/indexer/v1/supervisor%22,%1B[0m)
    "delta": "0:00:00.014053",
    "end": "2023-06-02 07:18:34.406570",
    "item": "rollup_summary_syncts",
    "rc": 7,
    "start": "2023-06-02 07:18:34.392517"
}

STDERR:

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 172.16.1.193 port 8090: Connection refused

MSG:

non-zero return code

failed: [172.16.1.193] (item=rollup_telemetry_audit_syncts) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": "curl -X POST -H \"Content-Type:application/json\" -d @/home/druid/rollup_telemetry_audit_syncts.json [http://172.16.1.193:8090/druid/indexer/v1/supervisor",](http://172.16.1.193:8090/druid/indexer/v1/supervisor%22,%1B[0m)
    "delta": "0:00:00.011369",
    "end": "2023-06-02 07:18:34.608830",
    "item": "rollup_telemetry_audit_syncts",
    "rc": 7,
    "start": "2023-06-02 07:18:34.597461"
}

STDERR:

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 172.16.1.193 port 8090: Connection refused

MSG:

non-zero return code

failed: [172.16.1.193] (item=rollup_telemetry_hourly_syncts) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": "curl -X POST -H \"Content-Type:application/json\" -d @/home/druid/rollup_telemetry_hourly_syncts.json [http://172.16.1.193:8090/druid/indexer/v1/supervisor",](http://172.16.1.193:8090/druid/indexer/v1/supervisor%22,%1B[0m)
    "delta": "0:00:00.020222",
    "end": "2023-06-02 07:18:34.859842",
    "item": "rollup_telemetry_hourly_syncts",
    "rc": 7,
    "start": "2023-06-02 07:18:34.839620"
}

STDERR:

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 172.16.1.193 port 8090: Connection refused

MSG:

non-zero return code

failed: [172.16.1.193] (item=rollup_telemetry_syncts) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": "curl -X POST -H \"Content-Type:application/json\" -d @/home/druid/rollup_telemetry_syncts.json [http://172.16.1.193:8090/druid/indexer/v1/supervisor",](http://172.16.1.193:8090/druid/indexer/v1/supervisor%22,%1B[0m)
    "delta": "0:00:00.016254",
    "end": "2023-06-02 07:18:35.103978",
    "item": "rollup_telemetry_syncts",
    "rc": 7,
    "start": "2023-06-02 07:18:35.087724"
}

STDERR:

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 172.16.1.193 port 8090: Connection refused

MSG:

non-zero return code

failed: [172.16.1.193] (item=rollup_tpd_hourly_syncts) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": "curl -X POST -H \"Content-Type:application/json\" -d @/home/druid/rollup_tpd_hourly_syncts.json [http://172.16.1.193:8090/druid/indexer/v1/supervisor",](http://172.16.1.193:8090/druid/indexer/v1/supervisor%22,%1B[0m)
    "delta": "0:00:00.022102",
    "end": "2023-06-02 07:18:35.398815",
    "item": "rollup_tpd_hourly_syncts",
    "rc": 7,
    "start": "2023-06-02 07:18:35.376713"
}

STDERR:

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 172.16.1.193 port 8090: Connection refused

MSG:

non-zero return code

PLAY RECAP *********************************************************************
172.16.1.193               : ok=4    changed=0    unreachable=0    failed=1    skipped=4    rescued=0    ignored=0   
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0   

[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (slack_notify)
[Pipeline] sh
+ git rev-parse --short HEAD
[Pipeline] sh
+ git name-rev --name-only HEAD
+ rev
+ + rev
cut -d / -f1
[Pipeline] sh
+ echo Deploy/alex/dev/DataPipeline/DruidIngestion
[Pipeline] echo
Could not find env specific Slack channel. Check for global slack channel..
[Pipeline] echo
Could not find slack environment variable. Skipping slack notification..
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (email_notify)
[Pipeline] sh
+ echo Deploy/alex/dev/DataPipeline/DruidIngestion
[Pipeline] echo
Could not find env specific email group. Check for global email group..
[Pipeline] echo
Could not find global email group variable. Skipping email notification..
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: script returned exit code 2
Finished: FAILURE

**Comment by tsprasath** [comment, marked as answer]
@srikanth-6520 ensure druid is running then execute this Jenkins job and check it. It has a dependency on druids.
